### PR TITLE
:sparkles: add QPS and burst for the agent components

### PR DESCRIPTION
**Pull request by zhiweiyin318**
## Related issue(s)

Fixes #

### CODE REVIEW

File: operator/v1/types_clustermanager.go [hunk 1]
```diff
@@ -114,6 +119,11 @@ type WorkConfiguration struct {
 	//  	he can set featuregate/Foo=false before upgrading. Let's say the cluster-admin wants featuregate/Foo=false.
 	// +optional
 	FeatureGates []FeatureGate `json:"featureGates,omitempty"`
+
+	// FlagOptions represents a list of flag options for the component binary.
+	// The key is the option name, the value is the option value.
+	// The FlagOptions can override all configured options including the default options and FeatureGates, etc.
+	FlagOptions map[string]string `json:"flagOptions,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]: would be better if it is a list with key/value. What if this and featuregates are all set?

Comment (zhiweiyin318) [reply operator/v1/types_clustermanager.go]: changed to a list with name/value.
My thought is that the configured flagOptions can override all default configuration including the featuregates, ClientCertExpirationSeconds and ClusterAnnotations, and the featuregates, ClientCertExpirationSeconds and ClusterAnnotations could be deprecated in the future.

File: operator/v1/types_clustermanager.go [hunk 2]
```diff
@@ -152,6 +170,13 @@ const (
 	FeatureGateModeTypeDisable FeatureGateModeType = "Disable"
 )
 
+type FlagOptions struct {
+	// Name is the flag option name of the component binary.
+	Name string `json:"name,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]: required, and need to do validation

File: operator/v1/types_clustermanager.go [hunk 3]
```diff
@@ -152,6 +170,13 @@ const (
...
+	// Value is the value of the flag option.
+	Value string `json:"value,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]: required

File: operator/v1/types_clustermanager.go [hunk 4]
```diff
@@ -126,6 +138,12 @@ type AddOnManagerConfiguration struct {
 	//  	he can set featuregate/Foo=false before upgrading. Let's say the cluster-admin wants featuregate/Foo=false.
 	// +optional
 	FeatureGates []FeatureGate `json:"featureGates,omitempty"`
+
+	// FlagOptions represents a list of flag options for the component binary.
+	// The key is the option name, the value is the option value.
+	// The FlagOptions can override all configured options including the default options and FeatureGates, etc.
+	// +optional
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]: set this to listMap

File: operator/v1/types_klusterlet.go [hunk 5]
```diff
@@ -157,6 +157,44 @@ type RegistrationConfiguration struct {
 	// ManagedCluster when creating only, other actors can update it afterwards.
 	// +optional
 	ClusterAnnotations map[string]string `json:"clusterAnnotations,omitempty"`
+
+	// KubeAPIQPS indicates the maximum QPS while talking with apiserver of hub cluster from the spoke cluster.
+	// If it is set empty, use the default value: 50
+	// +optional
+	// +kubebuilder:default:=50
+	// +kubebuilder:validation:Pattern=^(-?\d+)(.\d+)?$
+	KubeAPIQPS string `json:"qps,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]: the json field should be also kubeAPIQPS, why it is not int32 with max?

Comment (zhiweiyin318) [reply operator/v1/types_klusterlet.go]: changed to kubeAPIQPS. the QPS is float, but float is not suggested to be used in API, so use string here and added a validation.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**